src/broker/amqp.rs

```diff
@@ -301,13 +301,22 @@ impl Broker for AMQPBroker {
     }
 
     async fn close(&self) -> Result<(), BrokerError> {
-        // 320 reply-code = "connection-forced", operator intervened.
-        // For reference see https://www.rabbitmq.com/amqp-0-9-1-reference.html#domain.reply-code
+        let consume_channel = self.consume_channel.write().await;
+        let produce_channel = self.produce_channel.write().await;
         let conn = self.conn.lock().await;
+
         if conn.status().connected() {
-            debug!("Closing connection...");
+            debug!("Closing all channels and connections...");
+            consume_channel
+                .basic_cancel("", lapin::options::BasicCancelOptions::default())
+                .await?;
```

Comment:
Replacing this with `tokio::time::sleep(tokio::time::Duration::from_millis(100)).await` also prevents the error, which is why I'm skeptical.

Reply:
I'm surprised this basic_cancel even works given you pass it an empty consumer tag. If I was the server I'd throw you an error for this